app/admin/practices.rb

```diff
@@ -112,10 +50,33 @@
   filter :support_network_email
 
   controller do
-    def edit
-      redirect_to edit_practice_path(resource)
+    before_create do |practice|
+      if params[:user_email].present?
+        set_practice_user(practice)
+      end
+    end
+
+    before_update do |practice|
+      set_practice_user(practice) if params[:user_email].present?
+      practice.user = nil unless params[:user_email].present?
     end
 
+    def set_practice_user(practice)
+      email = params[:user_email]
+      user = User.find_by(email: email)
+
+      # create a new user if they do not exist
+      user = User.new(email: email) if user.blank?
+
+      # set the user's attributes based on ldap entry
+      user.skip_password_validation = true
+      user.skip_va_validation = true
```

Comment:
why do we want to set `skip_va_validation` to `true` here? Not sure what ldap is so this may contribute to my confusion. 😅

Reply:
so that we can make the `user`, haha
this is future-proofing for making the site public,
there is a requirement when you sign up that it has to be a va email.
but an admin can invite non-va emails, and this is how to do it.